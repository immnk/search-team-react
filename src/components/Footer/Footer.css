.root {
}

.footerImg {
  background: #fff url('./footer.png');
  background-position: 0 -175px;
  background-size: cover;
  height: 300px;
  width: 100%;
}

.wrapper {
  padding: 2rem;

  h3 {
    text-align: center;
    font-size: 3rem;
    font-weight: 100;
    font-family: 'Lobster', cursive;
  }
}

.linkWrapper {
  display: flex;
  justify-content: space-between;
  width: 720px;
  margin: 0 auto;

  @media screen and (max-width: 540px) {
    width: 100%;
  }
}

.aside {
  dd {
    margin-bottom: 1.2rem;
  }
}
