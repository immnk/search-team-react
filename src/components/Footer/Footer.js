import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import css from './Footer.css';

const Footer = ({ className }) => {
  return (
    <section className={classNames(css.root, className)}>
      <figure className={css.footerImg} />
      <section className={css.wrapper}>
        <h3>Developed by Mani</h3>
        <div className={css.linkWrapper}>
          <aside className={css.aside}>
            <dl>
              <dt>Source Code</dt>
              <dd>
                <a
                  href="https://github.com/immnk/search-team-react"
                  target="_blank"
                  rel="noreferrer"
                >
                  https://github.com/immnk/search-team-react
                </a>
              </dd>
              <dt>Hosted on</dt>
              <dd>
                <a href="https://serene-knuth-d39e99.netlify.app/" target="_blank" rel="noreferrer">
                  https://serene-knuth-d39e99.netlify.app/
                </a>
              </dd>
            </dl>
          </aside>
          <aside className={css.aside}>
            <dl>
              <dt>email me</dt>
              <dd>
                <a href="mailto:im.manikanta@gmail.com">im.manikanta@gmail.com</a>
              </dd>
              <dt>Call me</dt>
              <dd>
                <a href="tel:+6581214525">+65-81214525</a>
              </dd>
            </dl>
          </aside>
        </div>
      </section>
    </section>
  );
};

Footer.defaultProps = {
  className: undefined
};

Footer.propTypes = {
  className: PropTypes.string
};

export default Footer;
