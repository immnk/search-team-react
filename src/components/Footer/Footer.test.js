import React from 'react';
import { shallow } from 'enzyme';

import Footer from './Footer';

const renderFooter = (props) => {
  const defaultProps = {
    className: 'my-classname'
  };

  return <Footer {...defaultProps} {...props} />;
};

const shallowRenderFooter = (props) => shallow(renderFooter(props));

describe('Test Footer', () => {
  it('render app', () => {
    const actual = shallowRenderFooter();
    expect(actual).toMatchSnapshot();
  });
});
