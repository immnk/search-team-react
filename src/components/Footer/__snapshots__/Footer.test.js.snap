// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer render app 1`] = `
<section
  className="root my-classname"
>
  <div
    className="container"
  >
    <h3
      className="header"
    >
      Footer is here!
    </h3>
    <p
      className="description"
    >
      Some description is here
    </p>
  </div>
</section>
`;
